lib/n_plus_one_control/rspec/matchers/perform_linear_number_of_queries.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# rubocop:disable  Metrics/BlockLength
+::RSpec::Matchers.define :perform_linear_number_of_queries do |slope: 1|
+  supports_block_expectations
+
+  chain :with_scale_factors do |*factors|
+    @factors = factors
+  end
+
+  chain :with_warming_up do
+    @warmup = true
+  end
```

Comment:
I added both `with_scale_factors` and `with_warming_up` as these apply to this matcher too. I wasn't very sure about whether `matching` is needed.

Reply:
Yeah, query filtering (`matching`) could be useful here as well. Let's add it.